[peasantbird] iP

peasantbird opened the pull request:
### Bert
> "Your mind is for having ideas, not holding them." - David Allen [(source)](https://dansilvestre.com/productivity-quotes)

Bert frees your mind of having to remember things you need to do. It's
- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,
1. download it from [here](https://github.com/peasantbird/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

peasantbird committed:
Change Duke.java to Bert.java

peasantbird committed:
Add echo feature

peasantbird committed:
Add list and add features

peasantbird committed:
Add Task.java, and mark and unmark feature to Bert.java

peasantbird committed:
Add ToDo.java, Deadline.java, Event.java and support for tracking these tasks in Bert.java

peasantbird committed:
Test Bert.java

peasantbird committed:
Add error handling

peasantbird committed:
Add delete functionality

peasantbird committed:
Update tests

peasantbird committed:
Update UI and add error testing

peasantbird committed:
Add save tasks feature

peasantbird committed:
Clean up code and add new constructors to tasks

peasantbird committed:
Add load task feature

peasantbird committed:
Merge branch 'branch-Level-7'

lilozz2 reviewed (commented):
Was a lot cleaner than my own haha. Had extensive documentation which was something I really appreciate.

Overall code is clean and easy to read especially with the intuitive function names e.g. ensureTaskFileExists();
Comments were given everywhere, which was helpful in understanding the code
Test cases were sufficient and handled a majority of the cases

The only issue that I had was with the while loop and switch statements in the main function which I felt was doing too much and could have been better compartmentalised. The indentation was also going in too deep and starting to become like an arrow head.

iyioon commented on src/main/java/Bert.java:
```diff
@@ -0,0 +1,354 @@
+     */
+    private static void introduce() {
+        System.out.println(
+                "____________________________________________________________\n" +
```
Looks that the underbar line is being reused. Would it be simpler to abstract this such as having a UI do it?

iyioon commented on src/main/java/Bert.java:
```diff
@@ -0,0 +1,354 @@
+     *
+     * @param index The index of a task on the list
+     */
+    private static void mark(String index) {
```
Can you explain the reason behind receiving index as a String?

iyioon commented on src/main/java/Bert.java:
```diff
@@ -0,0 +1,354 @@
+     */
+    private static void unmark(String index) {
+        int i = Integer.parseInt(index) - 1;
+        Task t = al.get(i);
```
How about using `Task task = al.get(i)` for the variable name?

iyioon commented on src/main/java/Bert.java:
```diff
@@ -0,0 +1,354 @@
+            Scanner sc = new Scanner(file);
+            while (sc.hasNext()) {
+                String task = sc.nextLine();
+                switch (task.charAt(0)) {
```
You can follow the [Java Coding Standard](https://se-education.org/guides/conventions/java/intermediate.html#:~:text=The%20switch%20statement%20should%20have%20the%20following%20form%3A) for the switch statement.

iyioon reviewed (commented):
LGTM. Some naming can be improved as well as abstractions.

peasantbird committed:
Store dates as Localdate instead of String in deadline & event tasks

peasantbird committed:
Update tests to test date functionality

peasantbird committed:
Merge branch 'branch-Level-8'

peasantbird committed:
Refactor code to make the code more OOP

peasantbird committed:
Merge branch 'branch-A-MoreOOP'

peasantbird committed:
Refactor code into packages

peasantbird committed:
Merge branch 'branch-A-Packages'

peasantbird committed:
Merge branch 'add-gradle-support'

peasantbird committed:
Set up gradle to run Bert

peasantbird committed:
Add JUnit tests for TaskList and Parser

peasantbird committed:
Add JavaDoc comments to non-private classes/methods

peasantbird committed:
Merge branch 'branch-A-JavaDoc'

peasantbird committed:
Change variable names to improve readability

peasantbird committed:
Edit code to follow Java coding standard

peasantbird committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/bert/parser/Parser.java

peasantbird committed:
Edit JavaDoc comments to follow Java coding standard

peasantbird committed:
Add find tasks feature

peasantbird committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/bert/parser/Parser.java
#	src/main/java/bert/tasks/TaskList.java

peasantbird committed:
Create the GUI for Bert with JavaFX

peasantbird committed:
Merge branch 'branch-Level-10'

peasantbird committed:
Add assertions

The current code lack assertions to document important assumptions
that should hold at various points in the code.

Adding assertions improves the code quality as the runtime
can verify assumptions about Bert's state at certain points.

Let's start by adding some assertions to key methods like
the functions for handling user input, loading tasks from save file,
and to verify the execution of certain commands.

peasantbird committed:
Merge pull request #2 from peasantbird/branch-A-Assertions

Add assertions

peasantbird committed:
Improve code quality

Change variable names in DeleteCommand, MarkCommand and UnmarkCommand
to improve readability.

Add JavaDocs to explain code in Bert and Launcher clearly.

Refactor code in the save method of the Storage class. After the
implementation of the GUI, the system output is not needed.
Throws an IOException, which can be caught by the ExitCommand
to produce a CommandResult containing the error.

peasantbird committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/bert/commands/DeleteCommand.java
#	src/main/java/bert/commands/MarkCommand.java
#	src/main/java/bert/commands/UnmarkCommand.java

peasantbird committed:
Merge pull request #3 from peasantbird/branch-A-CodeQuality

Improve code quality

peasantbird committed:
Add sorting feature

peasantbird committed:
Merge branch 'branch-C-Sort'

peasantbird committed:
Give credit to code referred from AddressBook-Level2 and JavaFX tutorial part 4

peasantbird committed:
Add a representative screenshot of Bert

peasantbird committed:
Update README.md

peasantbird committed:
Update JUnit tests to ensure correctness

peasantbird committed:
Update README.md
